crates/red_knot_python_semantic/src/types.rs

```diff
@@ -702,3 +749,104 @@ pub struct TupleType<'db> {
     #[return_ref]
     elements: Box<[Type<'db>]>,
 }
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::needless_pass_by_value)]
```

Comment:
what's the reason for the clippy ignore here?

Reply:
It's probably because `Ty` is passed by value to test-functions that don't consume it

Reply:
Yeah; the alternative is sprinkling a bunch of annoying extra `&` in front of every Ty in every test case, which IMO harms readability for no actual gain. I can add a comment here explaining why this is here, I thought about doing that.

Reply:
I figured out how to remove the clippy suppression by changing the `to_type` method to a consuming `into_type` method instead (with some help from @MichaReiser to realize that in order to make this work I had to have `Ty::Union` keep a vec instead of a boxed array).